internal/monitors/docker/docker.go

```diff
@@ -225,9 +227,66 @@ func parseContainerEnvSlice(env []string) map[string]string {
 	return out
 }
 
+// EnableExtraGroups enables extra metrics that were individually turned on
+// by ExtraMetrics/ExtraGroups configuration
+func EnableExtraGroups(groupEnableMap map[string]bool, extraMetrics, extraGroups []string) EnhancedMetricsConfig {
+	groupNames := []string{groupBlkio, groupCPU, groupMemory, groupNetwork}
+
+	for _, group := range extraGroups {
+		for _, gn := range groupNames {
+			if group == gn {
+				groupEnableMap[gn] = true
+			}
+		}
+	}
+	for _, metric := range extraMetrics {
```

Comment:
Ideally this would be a helper method somewhere (either directly in generated code or common package that takes generated variables as argumnets) and also ideally we could do the check in O(1) for each metric, esp for monitors with lots of groups like elasticsearch. @jaysfx

Reply:
What do we think about having auto-generated `metricGroupMap` as well as `groupMetricsMap`?

Reply:
What we really should do, IMO, is to add `MetricInfo.Group` so we can get a group name for a metric in O(1)

Reply:
> What we really should do, IMO, is to add `MetricInfo.Group` so we can get a group name for a metric in O(1). `MetricInfo` looks like a place we would have this kind of info. @keitwb @jaysfx

yeah that seems like the easiest option.

Reply:
If we want to be exhaustive about this, we also need to handle the case where `extraMetrics` has a glob/regex in it by iterating every metric that might match it.  @jaysfx

Reply:
Yeah adding `Group` to `MetricInfo` makes sense.

I improved the enabled metrics list in the [this PR](https://github.com/signalfx/signalfx-agent/pull/743/files#diff-71a16ae4b4cab4ccd7b59767ef22b69bR358) (previously called expvar). It runs all the known metrics across the filterset (which includes both extra metrics and excluded datapoints). This would be ideal to use I think since in addition to handling globs if datapoints have been excluded there's no need to collect those metrics/enable relevant groups.

So maybe this computed `enabledMetrics` can be available to `Configure` (set it on the `MonitorConfig` struct or monitor itself?)